tools/swift-refactor/swift-refactor.cpp

```diff
@@ -402,15 +406,26 @@ int main(int argc, char *argv[]) {
   switch (options::DumpIn) {
   case options::DumpType::REWRITTEN:
     pConsumer.reset(new SourceEditOutputConsumer(SF->getASTContext().SourceMgr,
-                                                 BufferID,
-                                                 llvm::outs()));
+                                                 BufferID, llvm::outs()));
     break;
   case options::DumpType::JSON:
     pConsumer.reset(new SourceEditJsonConsumer(llvm::outs()));
     break;
-  case options::DumpType::TEXT:
-    pConsumer.reset(new SourceEditTextConsumer(llvm::outs()));
-    break;
+  case options::DumpType::TEXT: {
+    if (options::RewrittenOutputFile.empty()) {
+      pConsumer.reset(new SourceEditTextConsumer(llvm::outs()));
+    } else {
+      std::error_code EC;
+      static llvm::raw_fd_ostream FileStream(options::RewrittenOutputFile, EC,
+                                             llvm::sys::fs::OF_None);
+      if (FileStream.has_error() || EC) {
+        llvm::errs() << "Could not open rewritten output file";
+        return 1;
+      }
+      pConsumer.reset(new DuplicatingSourceEditConsumer(
+          llvm::outs(), SF->getASTContext().SourceMgr, BufferID, FileStream));
+    }
+  } break;
```

Comment:
Small stylistic nitpick: Could you remove the curly braces after the `case options::DumpType::TEXT:`?

```suggestion
  case options::DumpType::TEXT:
    if (options::RewrittenOutputFile.empty()) {
      pConsumer.reset(new SourceEditTextConsumer(llvm::outs()));
    } else {
      std::error_code EC;
      static llvm::raw_fd_ostream FileStream(options::RewrittenOutputFile, EC,
                                             llvm::sys::fs::OF_None);
      if (FileStream.has_error() || EC) {
        llvm::errs() << "Could not open rewritten output file";
        return 1;
      }
      pConsumer.reset(new DuplicatingSourceEditConsumer(
          llvm::outs(), SF->getASTContext().SourceMgr, BufferID, FileStream));
    }
    break;
```